app/src/main/java/org/oppia/app/player/audio/AudioFragmentPresenter.kt

```diff
@@ -84,10 +99,46 @@ class AudioFragmentPresenter @Inject constructor(
       it.audioFragment = fragment as AudioFragment
       it.lifecycleOwner = fragment
     }
-
+    subscribeToProfileLiveData()
     return binding.root
   }
 
+  private val profileLiveData: LiveData<Profile> by lazy {
+    getProfileData()
+  }
+
+  private fun getProfileData(): LiveData<Profile> {
+    return Transformations.map(
+      profileManagementController.getProfile(profileId),
+      ::processGetProfileResult
+    )
+  }
+
+  private fun subscribeToProfileLiveData() {
+    profileLiveData.observe(activity, Observer<Profile> { result ->
+      audioLanguage = result.audioLanguage
+      languageSelected(getAudioLanguage(audioLanguage))
+    })
+  }
+
+  /** Gets language code by [AudioLanguage]. */
+  fun getAudioLanguage(audioLanguage: AudioLanguage): String {
```

Comment:
Can this be private? It doesn't seem to be accessed outside this file, so prefer to minimize visibility.

Reply:
Done